posts/bksahus-gsoc-wrap-up.rst

```diff
@@ -0,0 +1,166 @@
+       tmp_arg1 = a
+       tmp_arg2 = b
+       tmp_arg3 = c
+       ...
+        tmp_iter_1 = iter(tmp_arg1)
+       tmp_iter_2 = iter(tmp_arg2)
+       tmp_iter_3 = iter(tmp_arg3)
+       ...
+        # could be more
+       tmp_result = []
+       try:
+           while 1:
+               tmp_result.append(
+                   (
+                        next(tmp_iter_1),
+                        next(tmp_iter_2),
+                        next(tmp_iter_3),
+                        ...
+                   )
+                )
+          except StopIteration:
+              pass
+        return tmp_result
+        
+   * `zip` for Python 3 needs a new node that calls the `zip` because unlike `zip` in Python 2, `zip` in Python 3 returns a    `zipobject`.     
+
+Test suite
+==========
+
+* Search mode "All": `PR #378 <https://github.com/Nuitka/Nuitka/pull/378>`
+   * In the test suite, I added a new search mode "all" that will test all the modules and return all the results at once.
+   Example::
+      
+      $ ./tests/basics/run_all.py all
+         Using concrete python 2.7.12 on x86_64
+         Comparing output of 'Asserts.py' using '/usr/bin/python' with flags silent, expect_success, remove_output,     recurse_all, original_file, cpython_cache, plugin_enable:pylint-warnings ...
+         .
+         .
+         .
+         .
+         Total 0 error(s) found.
+
+   
+   
+* Search mode "Only": `PR #333 <https://github.com/Nuitka/Nuitka/pull/333>`
+   * Added "only" search mode to test only a single module.
+   Example::
+   
+      $ ./tests/basics/run_all.py only BuiltinsTest.py
+         Using concrete python 2.7.12 on x86_64
+         Skipping Asserts.py
+         Skipping Assignments.py
+         Skipping BigConstants.py
+         Skipping Branching.py
+         Skipping BuiltinOverload.py
+         Skipping BuiltinSuper.py
+         Comparing output of 'BuiltinsTest.py' using '/usr/bin/python' with flags silent, expect_success, remove_output, recurse_all, original_file, cpython_cache, plugin_enable:pylint-warnings ...
+
+   
+* Reported and closed Issue #334: `PR #336 <https://github.com/Nuitka/Nuitka/pull/336>`
+   * Fixed the autoformat bug by reading and writing the files in bytes instead of string.
+
+Documentation
+=============
+* Nuitka git work flow: `PR #485 https://github.com/Nuitka/Nuitka/pull/485`
+
+And other minor doc fixes are added with their respective pull requests.
+
+What I learned
+==============
+* Learned the software engineering principles and how to keep my work clean.
+* I also learned how to effectively use software designing principles like DRY and KISS.
+* Got exposed to Nuitka internals which helped me to better understand how compilers in general work.
+* Explored how CPython works internally.
+* Got some great advice from Mentor Hayen about starting my professional career in Software engineering.
+
+Overall, it was a great experience to be a part of Nuitka :) 
+
+Yours, :raw-html:`<br />`
```

Comment:
Out of curiousity, I looked up, how to do it.

Check out http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html and "line blocks".

I think the way to do it is this then:
```
| Yours,
| Kay
```

I will try this out on my existing posts eventually (works this way). But you can leave it like this, I am not yet sure if links work in there and it's only a style/knowledge issue.

I like btw how you made your name a link.